addons/controllers/utils/utils.go

```diff
@@ -134,3 +135,57 @@ func TryParseClusterVariableString(ctx context.Context, cluster *clusterapiv1bet
 	}
 	return res
 }
+
+// GetOwnerCluster returns cluster owner for the object.
+// Returns cluster=nil and err if no cluster can be determined as owner.
+// Returns cluster object with only name and namespace  and err,
+// if details for the cluster determined to be an woner can not be found.
+func GetOwnerCluster(ctx context.Context, k8sClient client.Client, k8sObject client.Object, namespace, pkgRefName string) (*clusterapiv1beta1.Cluster, error) {
+	logger := log.FromContext(ctx)
+	cluster := &clusterapiv1beta1.Cluster{}
+	clusterName := ""
+
+	for _, ownerRef := range k8sObject.GetOwnerReferences() {
+		if strings.EqualFold(ownerRef.Kind, constants.ClusterKind) {
+			clusterName = ownerRef.Name
+			break
+		}
+	}
+	if clusterName != "" {
+		cluster.Name = clusterName
+		cluster.Namespace = namespace
+	} else {
+		// Means no cluster listed
+		// look at object name and match against patern of {clustern-name}-{packagerefname}-package
```

Comment:
This commet (booooo ) is wrong. We don't check against the patter shown. We can check against whatever pattern GenratePackage will return which in this case is {clustern-name}-{packagerefname}-package